ndk/src/media/media_codec.rs

```diff
@@ -538,3 +774,39 @@ pub enum DequeuedOutputBufferInfoResult<'a> {
     OutputFormatChanged,
     OutputBuffersChanged,
 }
+
+#[derive(Copy, Clone, Debug)]
+pub struct BufferInfo {
+    inner: ffi::AMediaCodecBufferInfo,
+}
```

Comment:
Should `OutputBuffer` contain (and `Deref` to) this so that it also has consistent access to all fields?

Reply:
This feels somewhat abusive of the `Deref` trait. `OutputBuffer` is not really a smart pointer[^1] for `BufferInfo` (if anything, it's a smart pointer for `&[u8]`). I'd prefer to add a `pub fn info(&self) -> &BufferInfo` method to `OutputBuffer`, but this would imply dropping the existing `flags` and `presentation_time_us` methods, which would obviously be a breaking change.

[^1]:  [`Deref` should only be implemented for smart pointers to avoid confusion.](https://doc.rust-lang.org/std/ops/trait.Deref.html)

Reply:
Let's have `fn info()`, totally fine to break the original API here!

Reply:
FYI we'll do a breaking release soon™ to match up with `winit` (a major consumer of the `ndk` crate), and any (breaking) cleanups we could get done to `MediaCodec` the better, so you're more than welcome to.